tensorflow/python/ops/linalg_ops.py

```diff
@@ -539,12 +538,23 @@ def norm(tensor,
 
   with ops.name_scope(name, 'norm', [tensor]):
     tensor = ops.convert_to_tensor(tensor)
+    rank = len(tensor.get_shape().as_list())
```

Comment:
This will not work if the rank is unknown at graph construction time. Please move this inside the special case in line 545 where you actually need this. Then change it to use tensorflow ops (tf.rank, tf.range etc., concat) to construct the permutation arrays at runtime instead.